bitcoin/src/network/constants.rs

```diff
@@ -92,22 +87,111 @@ impl Network {
     /// # Examples
     ///
     /// ```rust
-    /// use bitcoin::network::constants::Network;
+    /// use bitcoin::network::constants::{Network, Magic};
     ///
     /// let network = Network::Bitcoin;
-    /// assert_eq!(network.magic(), 0xD9B4BEF9);
+    /// assert_eq!(network.magic(), Magic::from_bytes([0xF9, 0xBE, 0xB4, 0xD9]));
     /// ```
-    pub fn magic(self) -> u32 {
-        // Note: any new entries here must be added to `from_magic` above
-        match self {
-            Network::Bitcoin => 0xD9B4BEF9,
-            Network::Testnet => 0x0709110B,
-            Network::Signet  => 0x40CF030A,
-            Network::Regtest => 0xDAB5BFFA,
+    pub fn magic(self) -> Magic {
+        Magic::from(self)
+    }
+}
+
+/// Network magic bytes to identify the cryptocurrency network the message was intended for.
+#[derive(Copy, PartialEq, Eq, PartialOrd, Ord, Clone, Hash)]
+pub struct Magic([u8; 4]);
+
+impl Magic {
+    /// Bitcoin mainnet network magic bytes.
+    pub const BITCOIN: Self = Self([0xF9, 0xBE, 0xB4, 0xD9]);
+    /// Bitcoin testnet network magic bytes.
+    pub const TESTNET: Self = Self([0x0B, 0x11, 0x09, 0x07]);
+    /// Bitcoin signet network magic bytes.
+    pub const SIGNET: Self = Self([0x0A, 0x03, 0xCF, 0x40]);
+    /// Bitcoin regtest network magic bytes.
+    pub const REGTEST: Self = Self([0xFA, 0xBF, 0xB5, 0xDA]);
```

Comment:
I just realized these have the reverse order compared to those above. Is this correct/intended?

Reply:
What do you mean by reverse order?
The bytes are in network order (or so I believe).

Reply:
`u32` parses as big endian and network order is presumably also big endian? I may be misremembering, I'll have to check later.

Reply:
These are little endian ("network byte order" is big endian [0] but Bitcoin consensus network order for a u32 is little endian (from looking at our code).

So these arrays are little endian, which is correct but the hex digits they replace were big endian (because that's how we write digits) so they are "reversed" as @Kixunil says, but the reversal is correct - phew!

[0] https://www.ibm.com/docs/en/zos/2.4.0?topic=hosts-network-byte-order

Reply:
TIL: Bitcoin network order is little-endian except for hashes which are big-endian on the wire bit displayed in little-endian. (but only for some hashes, right? I've no idea why some are dispalyed backwards or how those that are were picked and not the others? Anyone know?)

Reply:
Hashes do not have endianness.

Blockhashes and TXIDs are displayed backward because satoshi was writing in a weakly-typed dialect of C++ which caused them to be reversed in a series of type conversions. The hashes that were added after satoshi left, I believe, were not reversed. But it's pretty ad-hoc.

Reply:
> Hashes do not have endianness.

If we write a hash as a hex string "0xblahblahblah" then this can be put in memory, and on the wire, either big-endian or little-endian, right? One has to know the relation between the display and the wire format. Do you mean that we just don't typically refer to that as "endianness"?

Reply:
Oh I think I get it now, a hash is just a byte array so it does not have endianness, is that right?

Reply:
Oh, crap, that's a mess! So Core displays magic in the reversed order as it's on the wire? We should document it.

Reply:
I don't think the network magic is ever displayed under any circumstances.

Reply:
https://github.com/romanz/electrs/pull/768